src/arcaflow_plugin_sdk/schema.py

```diff
@@ -6787,12 +6787,14 @@ def _resolve_list_annotation(
                 "List type without item type definition encountered, please"
                 " declare your lists like this: typing.List[str]",
             )
+        sub_type = args[0]
+
         new_path = list(path)
         new_path.append("items")
         try:
             return ListType(
                 cls._resolve_abstract_type(
-                    args[0], type_hints, tuple(new_path), scope
+                    sub_type, sub_type, tuple(new_path), scope
```

Comment:
Isn't `type_hints` unused now and can be removed from the class input?

Reply:
There are a few instances where that is the case. One argument for keeping it is that it's the standard set of parameters for all type resolution calls.

Reply:
I'm not a fan of dead code, but I can understand keeping the parameters consistent across the calls.